### PR TITLE
Document the need for disabling BB if building without network access

### LINKED ISSUE #33569: julia-1.2.0-full.tar.gz does not build behind firewall
As This worked fine when I downloaded and built Julia 1.0 from source.  However, when I download the julia-1.2.0-full.tar.gz file from https://github.com/JuliaLang/julia/releases/download/v1.2.0/julia-1.2.0-full.tar.gz, I can't get it to build on an isolated network.  Currently the steps I follow are:

% tar -xvf julia-1.2.0-full.tar.gz
% make
Warning: git information unavailable; versioning information limited
Warning: git information unavailable; versioning information limited
Warning: git information unavailable; versioning information limited
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed connect to cache.julialang.org:443; Connection refused
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed connect to github.com:443; Connection refused
make[1]: *** [/home/user1/julia-1.2.0/deps/srccache/SuiteSparse.v5.4.0-2.x86_64-linux-gnu-gcc4.tar.gz] Error 7
make: *** [julia-deps] Error 2

OS:  RHEL 7.6 fully patched.

Any help would be greatly appreciated.

### PR DESCRIPTION
**Pull request by ViralBShah**
Fix #33569

### CODE REVIEW

File: doc/build/build.md [hunk 1]
```diff
@@ -17,8 +17,8 @@ variables.
 When compiled the first time, the build will automatically download
 pre-built [external
 dependencies](#required-build-tools-and-external-libraries). If you
-prefer to build all the dependencies on your own, add the following in
-`Make.user`
+prefer to build all the dependencies on your own, or are building on a system that cannot
+access the network during the build process add the following in `Make.user`:
```
Comment (DilumAluthge) [inline doc/build/build.md:21]:
```suggestion
access the network during the build process, add the following in `Make.user`:
```

**Review by DilumAluthge — APPROVED** [review]

**Comment by Keno** [comment]
This seems fine, but should we ship that as a default Make.user in the -full tarballs?

**Comment by DilumAluthge** [comment]
I think there was some disagreement about that in https://github.com/JuliaLang/julia/issues/33569

I think as a first step, it's sufficient to just update the documentation.

**Comment by Keno** [comment]
Well, I'm glad I'm consistent in my opinions at least.

**Pull request merged**